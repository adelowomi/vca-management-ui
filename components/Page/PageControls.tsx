import Link from 'next/link';
import * as React from 'react';

import { Btn } from './PageButtons';
import { H1, RowSection } from './PageStyledElements';
export interface PageControlsProps {
  onSubmit: (e: React.MouseEvent<HTMLButtonElement, MouseEvent>) => void;
  title: string;
  siteId: string | string[];
}

export const PageControls = ({
  onSubmit,
  title,
  siteId,
}: PageControlsProps) => {
  return (
    <>
      <RowSection className="justify-between">
        <H1>{title}</H1>
        <div className="flex flex-row justify-start space-x-5">
          <Btn color="primary" $bg="secondary" $px="sm">
            <Link href={`/sites/${siteId}/pages`}> Cancel</Link>
          </Btn>

          <Btn
            color="secondary"
            $bg="primary"
            $px="lg"
            onClick={onSubmit}
            type="submit"
          >
            Save & Publish
          </Btn>
        </div>
      </RowSection>
    </>
  );
};
