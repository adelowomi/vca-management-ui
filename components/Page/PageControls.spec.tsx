import 'jest-styled-components';

import React from 'react';

import { fireEvent, render, screen } from '../../test/testUtils';
import { PageControls } from './PageControls';

describe('PageHeaderStyle', () => {
  it('should render PageHeaderStyle snapshot', () => {
    const mockOnButtonClick = jest.fn();
    const { asFragment } = render(
      <PageControls
        siteId={'1234'}
        onSubmit={mockOnButtonClick}
        title="hello"
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render PageHeaderStyle snapshot', () => {
    const mockOnButtonClick = jest.fn();
    render(
      <PageControls
        siteId={'1234'}
        onSubmit={mockOnButtonClick}
        title="hello"
      />
    );
    fireEvent.click(screen.getByText('Save & Publish'));
    expect(mockOnButtonClick).toHaveBeenCalledTimes(1);
  });
});
