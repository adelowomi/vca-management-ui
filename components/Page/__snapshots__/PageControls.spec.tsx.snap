// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageHeaderStyle should render PageHeaderStyle snapshot 1`] = `
<DocumentFragment>
  .c0 {
  color: #828282;
  background: #d7d9da;
  padding: 15px 25px;
  border-radius: 2px;
  box-shadow: rgba(128,128,128,0.1);
  font-weight: 600;
  font-size: 13px;
}

.c0:focus {
  outline: none;
}

.c1 {
  color: #fff;
  background: #1890FF;
  padding: 15px 25px;
  border-radius: 2px;
  box-shadow: rgba(128,128,128,0.1);
  font-weight: 600;
  font-size: 13px;
}

.c1:focus {
  outline: none;
}

<div
    class="justify-between flex
flex-row"
  >
    <h1
      class="font-bold
text-3xl"
    >
      hello
    </h1>
    <div
      class="flex flex-row justify-start space-x-5"
    >
      <button
        class="c0"
        color="primary"
      >
        <a
          href="/sites/1234/pages"
        >
           Cancel
        </a>
      </button>
      <button
        class="c1"
        color="secondary"
        type="submit"
      >
        Save & Publish
      </button>
    </div>
  </div>
</DocumentFragment>
`;
